opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/TracerShim.java

```diff
@@ -110,6 +116,35 @@ public <C> SpanContext extract(Format<C> format, C carrier) {
 
   @Override
   public void close() {
+    if (isClosed) {
+      return;
+    }
     isClosed = true;
+
+    TracerProvider provider = maybeUnobfuscate(this.provider);
+    if (provider instanceof Closeable) {
+      try {
+        ((Closeable) provider).close();
+      } catch (RuntimeException | IOException e) {
+        logger.log(Level.INFO, "Exception caught while closing TracerProvider.", e);
+      }
+    }
+  }
+
+  private static TracerProvider maybeUnobfuscate(TracerProvider tracerProvider) {
```

Comment:
I'm confused by this method.
* in what sense is a provider "obfuscated"?
* if the class name equals to ObfuscatedTracerProvider, why not just cast to it and ask for the delegate?

Reply:
Check out the source for [OpenTelemetrySdk](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/all/src/main/java/io/opentelemetry/sdk/OpenTelemetrySdk.java#L138-L164).

`OpenTelemetrySdk` implements `OpenTelemetry`. The idea is that we don't want folks doing stuff like this:
```
OpenTelemetry openTelemetry = ...
SdkTracerProvider sdkTracerProvider = (SdkTracerProvider) openTelemetry.getTracerProvider();
sdkTracerProvider.shutdown();
```

If working in instrumentation code and all you have access to is `OpenTelemetry` (i.e. _not_ `OpenTelemetrySdk`), you shouldn't be able to cast and call SDK specific methods.

The obfuscation layer trys add friction to discourage this pattern. Though, as we see in this PR, its still possible via reflection.